Update rules_go and gazelle

kalbasit opened the pull request:
closes #130

achew22 commented on BUILD:
```diff
@@ -12,9 +12,7 @@
 
-load("@io_bazel_rules_go//go:def.bzl", "gazelle", "go_prefix")
-
-go_prefix("github.com/bazelbuild/bazel-watcher")
```
In order to remove this you have to add a `# gazelle:prefix github.com/example/project` in its place

kalbasit replied:
Actually, you don't need that because gazelle is already called with the prefix [here](https://github.com/bazelbuild/bazel-watcher/pull/137/files#diff-96239ee297e0a92ac6ff96a6bc434ef0L21).

achew22 commented:
In `.travis.yaml` you will also have to change the version of Bazel we are building with to something modern. WDYT about using 0.16.1?

kalbasit commented:
@achew22 I've updated the PR regarding your comments. Also because I'm running NixOS as my linux distribution, the `e2e` package does not work for me because it downloads a bazel binary, but binaries do not simply run on NixOS, can you please try running it on your end to make sure it worked?

```sh-session
$ bazel test //ibazel/...
INFO: Build options have changed, discarding analysis cache.
INFO: Analysed 11 targets (37 packages loaded).
INFO: Found 8 targets and 3 test targets...
INFO: Elapsed time: 1.505s, Critical Path: 0.10s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
//ibazel:go_default_test                                        (cached) PASSED in 0.2s
//ibazel/command:go_default_test                                (cached) PASSED in 3.3s
//ibazel/output_runner:go_default_test                          (cached) PASSED in 0.0s

Executed 0 out of 3 tests: 3 tests pass.
INFO: Build completed successfully, 1 total action
$ bazel build //ibazel/...
INFO: Analysed 11 targets (0 packages loaded).
INFO: Found 11 targets...
INFO: Elapsed time: 0.263s, Critical Path: 0.01s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
$ bazel run //:gazelle
INFO: Analysed target //:gazelle (23 packages loaded).
INFO: Found 1 target...
Target //:gazelle up-to-date:
  bazel-bin/gazelle.bash
INFO: Elapsed time: 0.338s, Critical Path: 0.01s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
$ bazel test //...
ERROR: /code/opensource/stories/bazel-watcher_update_rules_go/src/github.com/bazelbuild/bazel-watcher/e2e/live_reload/BUILD:5:1: no such package '@build_bazel_bazel_0_14_1//': Traceback (most recent call last):
        File "/home/kalbasit/.cache/bazel/_bazel_kalbasit/0b6768f9b6f944595468314c7a019f08/external/com_github_bazelbuild_bazel_integration_testing/tools/repositories.bzl", line 48
                _extract_bazel(rctx)
        File "/home/kalbasit/.cache/bazel/_bazel_kalbasit/0b6768f9b6f944595468314c7a019f08/external/com_github_bazelbuild_bazel_integration_testing/tools/repositories.bzl", line 39, in _extract_bazel
                fail(("`bazel version` returned non z...)))
`bazel version` returned non zero return code (256): java.io.IOException: Cannot run program "/home/kalbasit/.cache/bazel/_bazel_kalbasit/0b6768f9b6f944595468314c7a019f08/external/build_bazel_bazel_0_14_1/bin/bazel-real" (in directory "/home/kalbasit/.cache/bazel/_bazel_kalbasit/0b6768f9b6f944595468314c7a019f08/external/build_bazel_bazel_0_14_1"): error=2, No such file or directory and referenced by '//e2e/live_reload:go_default_test/bazel0.14.1'
ERROR: Analysis of target '//e2e/live_reload:go_default_test/bazel0.14.1' failed; build aborted: no such package '@build_bazel_bazel_0_14_1//': Traceback (most recent call last):
        File "/home/kalbasit/.cache/bazel/_bazel_kalbasit/0b6768f9b6f944595468314c7a019f08/external/com_github_bazelbuild_bazel_integration_testing/tools/repositories.bzl", line 48
                _extract_bazel(rctx)
        File "/home/kalbasit/.cache/bazel/_bazel_kalbasit/0b6768f9b6f944595468314c7a019f08/external/com_github_bazelbuild_bazel_integration_testing/tools/repositories.bzl", line 39, in _extract_bazel
                fail(("`bazel version` returned non z...)))
`bazel version` returned non zero return code (256): java.io.IOException: Cannot run program "/home/kalbasit/.cache/bazel/_bazel_kalbasit/0b6768f9b6f944595468314c7a019f08/external/build_bazel_bazel_0_14_1/bin/bazel-real" (in directory "/home/kalbasit/.cache/bazel/_bazel_kalbasit/0b6768f9b6f944595468314c7a019f08/external/build_bazel_bazel_0_14_1"): error=2, No such file or directory
INFO: Elapsed time: 2.182s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (18 packages loaded)
FAILED: Build did NOT complete successfully (18 packages loaded)
```

kalbasit commented:
@achew22 CI is green :grin:

achew22 commented:
@kalbasit woo! I'm getting the dreaded `This branch cannot be rebased due to conflicts` which is preventing the merge button. Do you have any idea what is causing that?

kalbasit commented:
I don't see that on my end, try reloading the page sometimes Github retains bad state. The PR is up to date with the current master.

achew22 commented:
Huh, I'm getting this

![image](https://user-images.githubusercontent.com/37206/45269388-a0040e00-b44a-11e8-9cf1-8e1b4417b3db.png)

kalbasit commented:
@achew22 ok let me squash the commits then, it might help.

achew22 commented:
Good idea. Thanks for doing that

kalbasit commented:
@achew22 I squashed the commits and added you as a `Co-Authored-By` which seems to have pissed off the googlebot, I hope you can still merge, can you please try again?

achew22 commented:
As the admin for the project I am happy to override the CLA bot given that it was me trying to be helpful that broke it. Let's make sure the tests pass then I'll merge you. Thanks so much for your contribution!

kalbasit commented:
Of course, thank you for your quick feedback!

achew22 commented:
I'm a quick reviewer when I'm not off the grid :wink:
